sdk/iothub/azure-resourcemanager-iothub/src/test/java/com/azure/resourcemanager/iothub/generated/IotHubManagerTests.java

```diff
@@ -0,0 +1,166 @@
+
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Random;
+
+public class IotHubManagerTests extends TestBase {
+
+    private static final Random RANDOM = new Random();
+    private static final Region REGION = Region.US_WEST2;
+    private String resourceGroupName = "rg" + randomPadding();
+    private IotHubManager iotHubManager;
+    private ResourceManager resourceManager;
+    private boolean testEnv;
+
+    @Override
+    public void beforeTest() {
+        final TokenCredential credential = new DefaultAzureCredentialBuilder().build();
+        final AzureProfile profile = new AzureProfile(AzureEnvironment.AZURE);
+
+        iotHubManager = IotHubManager
+            .configure()
+            .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC))
+            .authenticate(credential, profile);
+
+        resourceManager = ResourceManager
+            .configure()
+            .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC))
+            .authenticate(credential, profile)
+            .withDefaultSubscription();
+
+        // use AZURE_RESOURCE_GROUP_NAME if run in LIVE CI
+        String testResourceGroup = Configuration.getGlobalConfiguration().get("AZURE_RESOURCE_GROUP_NAME");
+        testEnv = !CoreUtils.isNullOrEmpty(testResourceGroup);
+        if (testEnv) {
+            resourceGroupName = testResourceGroup;
+        } else {
+            resourceManager.resourceGroups()
+                .define(resourceGroupName)
+                .withRegion(REGION)
+                .create();
+        }
+    }
+
+    @Override
+    protected void afterTest() {
+        if (!testEnv) {
+            resourceManager.resourceGroups().beginDeleteByName(resourceGroupName);
+        }
+    }
+
+    @Test
+    @DoNotRecord(skipInPlayback = true)
+    public void testIotHubDescription() {
+        IotHubDescription iotHubDescription = null;
+        try {
+            String descriptionName = "iotHub" + randomPadding();
+            // embedmeStart
+            iotHubDescription = iotHubManager.iotHubResources()
+                .define(descriptionName)
+                .withRegion(REGION)
+                .withExistingResourceGroup(resourceGroupName)
+                .withSku(new IotHubSkuInfo().withName(IotHubSku.F1).withCapacity(1L))
+                .withIdentity(new ArmIdentity().withType(ResourceIdentityType.NONE))
+                .withProperties(
+                    new IotHubProperties()
+                        .withEventHubEndpoints(
+                            new HashMap<String, EventHubProperties>() {
+                                {
+                                    put("events", new EventHubProperties()
+                                        .withRetentionTimeInDays(1L)
+                                        .withPartitionCount(2));
+                                }
+                            })
```

Comment:
please just use normal Java code for Map, I assume you know what to write?

```
Map<> someMap = new HashMap<>();
someMap.put(.., ..);

Reply:
In this case of only a single entry, use `Collections.singletonMap`

Reply:
Fixed in the new version.